src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,280 @@
 
-            if (memberLevels.any { it <= config.markBelowClassLevel }) {
-                slot highlight LorenzColor.YELLOW
-                continue
-            }
+    }
 
-            if (config.markMissingClass && memberClasses.none { it == selectedClass }) {
-                slot highlight LorenzColor.GREEN
+    @SubscribeEvent
+    fun onToolTipRender(event: LorenzToolTipEvent) {
+        if (!isEnabled()) return
+        if (!inInventory) return
+        val toolTip = toolTipMap[event.slot.slotIndex]
+        if (toolTip.isNullOrEmpty()) return
+        val oldToolTip = event.toolTip
+        for((index, line) in toolTip.withIndex()) {
+            if (index >= event.toolTip.size-1) {
+                event.toolTip.add(line)
+                continue
             }
+            if(oldToolTip[index] != line) event.toolTip[index + 1] = line
         }
     }
 
     @SubscribeEvent
-    fun onItemTooltip(event: LorenzToolTipEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-
-        val chestName = InventoryUtils.openInventoryName()
-        if (chestName != "Party Finder") return
-
-        val stack = event.itemStack
-
-        val classNames = mutableListOf("Healer", "Mage", "Berserk", "Archer", "Tank")
-        for ((index, line) in stack.getLore().withIndex()) {
-            classLevelPattern.matchMatcher(line) {
-                val playerName = group("playerName")
-                val className = group("className")
-                val level = group("level").toInt()
-                val color = getColor(level)
-                if (config.coloredClassLevel) event.toolTip[index + 1] = " §b$playerName§f: §e$className $color$level"
-                classNames.remove(className)
-            }
-        }
-        if (!config.showMissingClasses) return
-        if (stack.getLore().firstOrNull()?.removeColor()?.startsWith("Dungeon:") == false) return
-        if (classNames.contains(selectedClass)) selectedClass = "§a${selectedClass}§7"
-        event.toolTip.add("")
-        event.toolTip.add("§cMissing: §7" + createCommaSeparatedList(classNames))
+    fun onRenderItemTip(event: RenderInventoryItemTipEvent) {
+        if (!isEnabled()) return
+        if (!config.floorAsStackSize) return
+        if (floorStackSize[event.slot.slotIndex].isNullOrEmpty()) return
+        event.stackTip = floorStackSize[event.slot.slotIndex] ?: ""
     }
 
     @SubscribeEvent
-    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
-        event.move(2, "dungeon.partyFinderColoredClassLevel", "dungeon.partyFinder.coloredClassLevel")
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        if (!inInventory) return
+
+        event.gui.inventorySlots.inventorySlots
+            .filter { highlightParty.containsKey(it.slotNumber) }
+            .forEach { it highlight highlightParty[it.slotNumber]!! }
```

Comment:
dont use !! handle the null somehow

Reply:
this use is ok (In my opinion), since the !! is only here because the smart case doesn't work.

Reply:
soo should I change it or keep it?

Reply:
change it, hanni made a better solution in the other PR of yours